common/src/main/java/tachyon/security/PlainSaslServer.java

```diff
@@ -0,0 +1,151 @@
+  private String mAuthcid;
+  private boolean mCompleted;
+  private CallbackHandler mHandler;
+
+  PlainSaslServer(CallbackHandler handler) throws SaslException {
+    mHandler = handler;
+  }
+
+  @Override
+  public String getMechanismName() {
+    return "PLAIN";
+  }
+
+  @Override
+  public byte[] evaluateResponse(byte[] response) throws SaslException {
+    if (mCompleted) {
+      throw new IllegalStateException("PLAIN authentication has completed");
+    }
+    if (response == null) {
+      throw new IllegalArgumentException("Received null response");
+    }
+    try {
+      // parse the response
+      // message   = [authzid] UTF8NUL authcid UTF8NUL passwd'
+      // authzid may be empty,then the authzid = authcid
+      String payload;
+      try {
+        payload = new String(response, "UTF-8");
+      } catch (Exception e) {
+        throw new IllegalArgumentException("Received corrupt response", e);
+      }
+      String[] parts = payload.split("\u0000", 3);
+      // validate response
+      if (parts.length != 3) {
+        throw new IllegalArgumentException("Invalid message format, parts must contain 3 items");
+      }
+      String authzid = parts[0];
+      mAuthcid = parts[1];
+      String passwd = parts[2];
+      if (mAuthcid == null || mAuthcid.isEmpty()) {
+        throw new IllegalStateException("No authentication identity provided");
+      }
+      if (passwd == null || passwd.isEmpty()) {
+        throw new IllegalStateException("No password provided");
+      }
+      if (authzid == null || authzid.isEmpty()) { // authzid = authcid
+        authzid = mAuthcid;
+      }
+
+      NameCallback nameCallback = new NameCallback("User");
+      nameCallback.setName(mAuthcid);
+      PasswordCallback passwordCallback = new PasswordCallback("Password", false);
+      passwordCallback.setPassword(passwd.toCharArray());
+      AuthorizeCallback authCallback = new AuthorizeCallback(mAuthcid, authzid);
+
+      Callback[] cbList = {nameCallback, passwordCallback, authCallback};
+      mHandler.handle(cbList);
+      if (!authCallback.isAuthorized()) {
+        throw new SaslException("AuthorizeCallback authorized failure");
+      }
+    } catch (IOException ioe) {
+      throw new SaslException("Error validating the response", ioe);
+    } catch (UnsupportedCallbackException uce) {
+      throw new SaslException("Error validating the response", uce);
+    }
+    mCompleted = true;
+    return null;
+  }
+
+  @Override
+  public boolean isComplete() {
+    return mCompleted;
+  }
+
+  @Override
+  public String getAuthorizationID() {
+    throwIfNotComplete();
+    return mAuthcid;
+  }
+
+  @Override
+  public byte[] unwrap(byte[] incoming, int offset, int len) {
+    throwIfNotComplete();
+    throw new UnsupportedOperationException("PLAIN doesn't support wrap or unwrap operation");
+  }
+
+  @Override
+  public byte[] wrap(byte[] outgoing, int offset, int len) {
+    throwIfNotComplete();
```

Comment:
Similar: Given it's not supported anyway, maybe we should not `throwIfNotComplete()` ? (So that developer/tester learn it's unsupported more quickly)